infrastructure/sentry-subgraph/src/referee.ts

```diff
@@ -506,40 +543,211 @@ export function handleUnstakeV1(event: UnstakeV1): void {
 }
 
 export function handleNewPoolSubmission(event: NewPoolSubmissionEvent): void {
-  let poolSubmission = new PoolSubmission(event.params.challengeId.toHexString() + event.params.poolAddress.toHexString())
+
+  const challenge = Challenge.load(event.params.challengeId.toString())
+  if (!challenge) {
+    log.warning("Failed to find challenge handleNewPoolSubmission: keyID: " + event.params.challengeId.toString() + ", TX: " + event.transaction.hash.toHexString(), [])
+    return;
+  }
+
+  const pool = PoolInfo.load(event.params.poolAddress.toHexString())
+  if (!pool) {
+    log.warning("Failed to find pool handleNewPoolSubmission: poolAddress: " + event.params.poolAddress.toString() + ", TX: " + event.transaction.hash.toHexString(), [])
+    return;
+  }
+
+  let poolSubmission = new PoolSubmission(event.params.poolAddress.toHexString() + "_" + event.params.challengeId.toString())
   poolSubmission.challengeId = event.params.challengeId
+  poolSubmission.poolAddress = event.params.poolAddress
+  poolSubmission.challenge = challenge.id
+  poolSubmission.poolInfo = pool.id
   poolSubmission.stakedKeyCount = event.params.stakedKeys
   poolSubmission.winningKeyCount = event.params.winningKeys
   poolSubmission.claimedRewardsAmount = BigInt.fromI32(0)
-  poolSubmission.poolAddress = event.params.poolAddress
+  poolSubmission.createdTimestamp = event.block.timestamp
+  poolSubmission.createdTxHash = event.transaction.hash
+  poolSubmission.claimTimestamp = BigInt.fromI32(0)
+  poolSubmission.claimTxHash = Bytes.fromI32(0)
   poolSubmission.claimed = false
   poolSubmission.save()
+
+  let poolChallenges = PoolChallenge.load(event.params.poolAddress.toHexString() + "_" + event.params.challengeId.toString())
```

Comment:
Pool challenges redundant?

Reply:
They are with the poolsubmissions as they mirror the same relationship of pool to challenge.
However we have the data sync runtimes that will calculate the reward rates by pool challenge and extending pool challenge for pool submission info would break the naming convention as there are now Pool submissions in the Referee.

We should follow up on this with its own story to get rid of one of the two.